### PR TITLE
[1.0.0] Upgrade script to convert values.yaml to 1.0.0

### PR DESCRIPTION
**Pull request by samjsong**
###### Description

Bash script that takes existing `values.yaml` file and spits out `new.yaml`, compatible with `1.0.0`

**TODO** have to fix the URL and the docs links once those are ready

### CODE REVIEW

File: deploy/helm/sumologic/upgrade-1.0.0.sh [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+This script will automatically take the configurations of your existing values.yaml
+and return one that is compatible with v1.0.0.
+
+Usage:
+  ./upgrade-1.0.0.sh /path/to/values.yaml
+
+Returns:
+  new.yaml
+
+For more details, please refer to Migration steps and Changelog here: [link]
```
Comment (samjsong) [inline deploy/helm/sumologic/upgrade-1.0.0.sh]: todo: add a proper link once changelog is ready

File: deploy/helm/sumologic/upgrade-1.0.0.sh [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+if [ "$1" = "" ]
+then
+  echo "$MAN"
+  exit 1
+fi
+
+OLD_VALUES_YAML=$1
+
+URL=https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/846cae152f4ccdc9cc1a26e4a67af1f603ea83fa/deploy/helm/sumologic/values.yaml
```
Comment (samjsong) [inline deploy/helm/sumologic/upgrade-1.0.0.sh]: todo: change this URL to
```
https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/v1.0.0/deploy/helm/sumologic/values.yaml
```
once it exists

File: deploy/helm/sumologic/upgrade-1.0.0.sh [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+sumologic.sourceName
+sumologic.sourceCategory
+sumologic.sourceCategoryPrefix
+sumologic.sourceCategoryReplaceDash
+sumologic.addTimestamp
+sumologic.timestampKey
+sumologic.verifySsl
+sumologic.excludeContainerRegex
+sumologic.excludeHostRegex
+sumologic.excludeNamespaceRegex
```
[the remark this answers is not in the document]
Comment (samjsong) [reply deploy/helm/sumologic/upgrade-1.0.0.sh]: You're totally right, that was a miss on my part (those were added to the google doc later). Will fix, thanks Sakshi!

**Comment by samjsong** [comment]
I've made the change to update from `deployment` and `eventsDeployment` to `fluentd.statefulset` and `fluentd.eventsStatefulset`, but due to limitations of `yq` the code isn't very clean

---

I also realized we made some changes to the prometheus remote write regex config - this is both for cost reduction (removed metrics) and for kubernetes version support (renamed metrics). I couldn't figure out a way to do this using `yq` - while `yq` does support array indexing like
```
yq r values.yaml prometheus-operator.prometheus.prometheusSpec.remoteWrite[0].writeRelabelConfigs[0].regex
```
since this is index based, and customers had free reign to change their config in whatever way they wanted, I couldn't think of a way to update these regexes in a one-size-fits-all manner. Anybody have any ideas around this?

File: deploy/helm/sumologic/upgrade-1.0.0.sh [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+if [ "$1" = "" ]
+then
+  echo "$MAN"
+  exit 1
+fi
+
+OLD_VALUES_YAML=$1
+
+URL=https://raw.githubusercontent.com/SumoLogic/sumologic-kubernetes-collection/4bd5a737b0f9730e64c5d59f168420af73e81a9b/deploy/helm/sumologic/values.yaml
+curl -s $URL > new.yaml
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/upgrade-1.0.0.sh]: Maybe we could use link to release branch here?

File: deploy/helm/sumologic/upgrade-1.0.0.sh (hunk 1 above)
Comment (sumo-drosiek) [reply deploy/helm/sumologic/upgrade-1.0.0.sh]: Are we going to add info about requirements (`yq`) in this file or in `Migration steps`

**Comment by sumo-drosiek** [comment]
I run the script and have non positive output:

```
yq: -i/--in-place can only be used with -y/-Y
usage: yq [-h] [--yaml-output] [--yaml-roundtrip] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          jq_filter [files [files ...]]
yq: error: argument files: can't open 'sumologic.eventCollectionEnabled': [Errno 2] No such file or directory: 'sumologic.eventCollectionEnabled'
yq: -i/--in-place can only be used with filename arguments, not on standard input
yq: -i/--in-place can only be used with filename arguments, not on standard input
usage: yq [-h] [--yaml-output] [--yaml-roundtrip] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          jq_filter [files [files ...]]
yq: error: argument files: can't open 'sumologic.events.sourceCategory': [Errno 2] No such file or directory: 'sumologic.events.sourceCategory'
yq: -i/--in-place can only be used with filename arguments, not on standard input
yq: -i/--in-place can only be used with filename arguments, not on standard input
usage: yq [-h] [--yaml-output] [--yaml-roundtrip] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          jq_filter [files [files ...]]
yq: error: argument files: can't open 'sumologic.logFormat': [Errno 2] No such file or directory: 'sumologic.logFormat'
yq: -i/--in-place can only be used with filename arguments, not on standard input
yq: -i/--in-place can only be used with filename arguments, not on standard input
usage: yq [-h] [--yaml-output] [--yaml-roundtrip] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          jq_filter [files [files ...]]
```

**Comment by sumo-drosiek** [comment]
> since this is index based, and customers had free reign to change their config in whatever way they wanted, I couldn't think of a way to update these regexes in a one-size-fits-all manner. Anybody have any ideas around this?

I think reasonable way would be to generate placeholders and ask customers to update it  manually

**Comment by sumo-drosiek** [comment]
Mistery solved. I had another `yq` (`python-yq`) instead of
https://github.com/mikefarah/yq

**Comment by sumo-drosiek** [comment]
I did some tests with valid yq and seems that they are some changes we probably want to backport (prometheus regexes and falco should be disable by default) for values.yaml

**Comment by sumo-drosiek** [comment]
Also what about `sumologic.sourceHost` do we want to remove it?

**Comment by sumo-drosiek** [comment]
summon @samjsong :)

File: deploy/helm/sumologic/upgrade-1.0.0.sh (hunk 4 above)
Comment (samjsong) [reply deploy/helm/sumologic/upgrade-1.0.0.sh:27]: For sure, I left a TODO review comment to myself above but can update that now

**Comment by samjsong** [comment]
Thanks Dominik for the comments! Falco disable was a miss on my part - I will add that change. I will remove `sumologic.sourceHost` as well

For the prometheus regex changes, those were the ones I was concerned about - I couldn't think of a way to guarantee correctness, unless maybe we use `sed` or something for that as well 😓

File: deploy/helm/sumologic/upgrade-1.0.0.sh (hunk 1 above)
Comment (samjsong) [reply deploy/helm/sumologic/upgrade-1.0.0.sh:15]: I was thinking we could add that to the migration doc - but we could add that here as well

**Comment by samjsong** [comment]
Removed `sumologic.sourceHost` here: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/546

**Comment by sumo-drosiek** [comment]
@samjsong Great work 🔥 
I was thinking about backporting regexes from `1.0.0` to `0.17.1`, but I can also go through it tomorrow and write some unit tests 🤔 
Was this regex change related to prometheus update or something?

**Comment by samjsong** [comment]
Unfortunately the regex changes are a backwards breaking change since it will cause duplicate ingest of some metrics without the proper fluentd changes as well. so that is why we included it with 1.0 instead of 0.17

**Comment by perk-sumo** [comment]
Please consider this PR: https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/554

**Comment by samjsong** [comment]
Will close and reopen this PR since CI thinks something is wrong (manual changes to `fluentd-sumologic.yaml.tmpl`)

**Pull request closed without merging**